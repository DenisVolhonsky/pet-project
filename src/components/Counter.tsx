import { useState } from 'react';

interface CounterProps {
  children: React.ReactNode;
}

const Counter: React.FC<CounterProps> = ({ children }) => {
  const [result, setResult] = useState(0);

  const handleIncrease = () => {
    setResult((prevResult) => prevResult + 1);
  };

  const handleDecrise = () => {
    setResult((prevResult) => prevResult - 1);
  };

  return (
    <div>
      {children}
      <span>{result}</span>
      <button
        className="mt-2 bg-indigo-500 text-white p-1 mr-2 ml-2"
        onClick={handleIncrease}
      >
        +
      </button>
      <button
        className="mt-2 bg-indigo-500 text-white p-1"
        onClick={handleDecrise}
      >
        -
      </button>
    </div>
  );
};

export default Counter;
