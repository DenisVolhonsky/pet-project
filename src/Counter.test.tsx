import { render, screen, fireEvent } from '@testing-library/react';
import Counter from './components/Counter';

describe('Counter component', () => {
  it('renders children content correctly', () => {
    render(<Counter>Test Child</Counter>);

    // Проверка, что текст ребенка отображается
    expect(screen.getByText(/Test Child/i)).toBeInTheDocument();
  });

  it('renders initial counter value', () => {
    render(<Counter>Counter</Counter>);

    // Проверка, что начальное значение счетчика равно 0
    expect(screen.getByText('0')).toBeInTheDocument();
  });

  it('increases the counter value when + button is clicked', () => {
    render(<Counter>Counter</Counter>);

    const incrementButton = screen.getByText('+');

    // Клик по кнопке увеличения
    fireEvent.click(incrementButton);

    // Проверка, что значение счетчика увеличилось до 1
    expect(screen.getByText('1')).toBeInTheDocument();
  });

  it('decreases the counter value when - button is clicked', () => {
    render(<Counter>Counter</Counter>);

    const decrementButton = screen.getByText('-');

    // Клик по кнопке уменьшения
    fireEvent.click(decrementButton);

    // Проверка, что значение счетчика уменьшилось до -1
    expect(screen.getByText('-1')).toBeInTheDocument();
  });

  it('correctly handles multiple increments and decrements', () => {
    render(<Counter>Counter</Counter>);

    const incrementButton = screen.getByText('+');
    const decrementButton = screen.getByText('-');

    // Несколько кликов на увеличение
    fireEvent.click(incrementButton);
    fireEvent.click(incrementButton);

    // Значение должно быть 2
    expect(screen.getByText('2')).toBeInTheDocument();

    // Несколько кликов на уменьшение
    fireEvent.click(decrementButton);
    fireEvent.click(decrementButton);
    fireEvent.click(decrementButton);

    // Значение должно быть -1
    expect(screen.getByText('-1')).toBeInTheDocument();
  });
});

describe('Counter component snapshot', () => {
  it('should match the snapshot', () => {
    const { asFragment } = render(<Counter>Test Counter</Counter>);
    expect(asFragment()).toMatchSnapshot();
  });
});
