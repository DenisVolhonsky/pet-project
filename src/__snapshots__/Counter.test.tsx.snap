// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter component snapshot should match the snapshot 1`] = `
<DocumentFragment>
  <div>
    Test Counter
    <span>
      0
    </span>
    <button
      class="mt-2 bg-indigo-500 text-white p-1 mr-2 ml-2"
    >
      +
    </button>
    <button
      class="mt-2 bg-indigo-500 text-white p-1"
    >
      -
    </button>
  </div>
</DocumentFragment>
`;
